app/models/batch.rb

```diff
@@ -26,7 +26,7 @@ def self.entity_scope
 
   validate :date_produced_is_a_date
 
-  validates :isolate_name, uniqueness: { scope: :batch_number }
+  validate :isolate_name_batch_number_combination
```

Comment:
Case sensitivie option only applies for the attribute thats being validated, in this case `isolate_name`, but it does not hace any effect on the batch number. so the `:case_sensitive` option does not work for this case. 
As per discussed earlier, `batch_id` is not the best alternative because of the `id` column that the model already has, so we will be leaving `batch_number` until we make sure that this model does not get refactored and then think of a better attribute name.